controllers/component_controller.go

```diff
@@ -470,46 +480,51 @@ func (r *ComponentReconciler) generateGitops(ctx context.Context, req ctrl.Reque
 		return err
 	}
 
-	// Create a temp folder to create the gitops resources in
-	tempDir, err := ioutils.CreateTempPath(component.Name, r.AppFS)
-	if err != nil {
-		log.Error(err, "unable to create temp directory for GitOps resources due to error")
-		return fmt.Errorf("unable to create temp directory for GitOps resources due to error: %v", err)
-	}
-
-	deployAssociatedComponents, err := devfileParser.GetDeployComponents(compDevfileData)
-	if err != nil {
-		log.Error(err, "unable to get deploy components")
-		return err
-	}
+	// Determine if we're using a Kubernetes job for gitops generation, or generating locally
+	localGitopsGen := (r.AllowLocalGitopsGen && component.Annotations["allowLocalGitopsGen"] == "true") || (r.DoLocalGitOpsGen)
+	if localGitopsGen {
+		err = gitopsjoblib.GenerateGitopsBase(context.Background(), r.Client, *component, ioutils.NewFilesystem(), generate.GitOpsGenParams{
+			Generator: r.Generator,
+			Token:     r.GitToken,
+			RemoteURL: gitOpsURL,
+			Branch:    gitOpsBranch,
+			Context:   gitOpsContext,
+		})
+		if err != nil {
+			return err
+		}
+	} else {
+		gitopsJobConfig := gitopsjob.GitOpsJobConfig{
+			OperationType: gitopsjob.GenerateBase,
+			RepositoryURL: component.Status.GitOps.RepositoryURL,
+			ResourceName:  component.GetName(),
+			Branch:        gitOpsBranch,
+			Context:       gitOpsContext,
+		}
 
-	kubernetesResources, err := devfile.GetResourceFromDevfile(log, compDevfileData, deployAssociatedComponents, component.Name, component.Spec.ContainerImage)
-	if err != nil {
-		log.Error(err, "unable to get kubernetes resources from the devfile outerloop components")
-		return err
-	}
+		// Create a Kubernetes job (with random string appended to name to ensure uniqueness) to generate the GitOps resources
+		jobName := component.GetName()
+		if len(jobName) > 57 {
+			jobName = component.GetName()[0:56]
+		}
```

Comment:
something to ponder upon, if there are two components with super similar name and len > 57, there might be a name clash for the job. Do we want to suffix random chars?

Reply:
oops didnt see the next line